services/src/main/java/org/keycloak/organization/admin/resource/OrganizationIdentityProvidersResource.java

```diff
@@ -102,37 +96,40 @@ public Stream<IdentityProviderRepresentation> getIdentityProviders() {
     }
 
     @Path("{alias}")
-    public IdentityProviderResource getIdentityProvider(@PathParam("alias") String alias) {
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public IdentityProviderRepresentation getIdentityProvider(@PathParam("alias") String alias) {
         IdentityProviderModel broker = realm.getIdentityProviderByAlias(alias);
-        return new IdentityProviderResource(auth, realm, session, broker, adminEvent) {
-            @Override
-            public Response delete() {
-                Response response = super.delete();
 
-                if (organizationProvider.removeIdentityProvider(organization, broker)) {
-                    return response;
-                }
+        if (!isOrganizationBroker(broker)) {
+            throw ErrorResponse.error("Identity provider not found with the given alias", Status.NOT_FOUND);
+        }
 
-                throw ErrorResponse.error("Identity provider not associated with the organization", Status.BAD_REQUEST);
-            }
+        return ModelToRepresentation.toRepresentation(realm, broker);
+    }
 
-            @Override
-            public Response update(IdentityProviderRepresentation providerRep) {
-                if (organization.getIdentityProviders().noneMatch(model -> model.getInternalId().equals(providerRep.getInternalId()) || model.getAlias().equals(providerRep.getAlias()))) {
-                    return Response.status(Status.NOT_FOUND).build();
-                }
-                String domain = providerRep.getConfig().get(OrganizationModel.ORGANIZATION_DOMAIN_ATTRIBUTE);
+    @Path("{alias}")
+    @DELETE
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response delete(@PathParam("alias") String alias) {
+        IdentityProviderModel broker = realm.getIdentityProviderByAlias(alias);
 
-                if (domain != null && organization.getDomains().map(OrganizationDomainModel::getName).noneMatch(domain::equals)) {
-                    return Response.status(Status.BAD_REQUEST).build();
-                }
+        if (!isOrganizationBroker(broker)) {
+            throw ErrorResponse.error("Identity provider not found with the given alias", Status.NOT_FOUND);
```

Comment:
Shouldn't the message indicate that no provider with the specified alias is currently bound to the org? If you prefer we can create a follow up issue to revisit all messages returned from endpoints.

Reply:
Changed.